Source/AliveLibAO/Abe.cpp

```diff
@@ -2738,11 +2740,393 @@ void Abe::VOn_Tlv_Collision_421130(Path_TLV *pTlv)
+                const FP xPosToMidLiftPlatformDistance = FP_Abs(field_A8_xpos - liftPlatformXMidPoint);
+                if (xPosToMidLiftPlatformDistance < halfGrid)
+                {
+                    //AO exclusive - Abe only uses lift facing one side
+                    if (field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX))
+                    {
+                        field_FC_current_motion = eAbeStates::State_2_StandingTurn_426040;
+                        field_FE_next_state = eAbeStates::State_133_LiftGrabBegin_42EF20;
+                    }
+                    else
+                    {
+                        field_FC_current_motion = eAbeStates::State_133_LiftGrabBegin_42EF20;
+                    }
+                    return;
+                }
+            }
+        }
+
+        field_FC_current_motion = TryMountElum_42E600();
+        if (field_FC_current_motion != eAbeStates::State_0_Idle_423520)
+        {
+            return;
+        }
+        pTlv = gMap_507BA8.TLV_Get_At_446060(
+            nullptr,
+            field_A8_xpos,
+            field_AC_ypos,
+            field_A8_xpos,
+            field_AC_ypos
+        );
+
+        while (pTlv)
+        {
+            switch (pTlv->field_4_type)
+            {
+                case TlvTypes::Door_6:
+                {
+                    if (NearDoorIsOpen() && !field_10A_flags.Get(Flags_10A::e10A_Bit5_Electrocuted))
+                    {
+                        field_F0_pTlv = pTlv;
+                        field_110_state = 0;
+                        field_FC_current_motion = eAbeStates::State_156_DoorEnter_42D370;
+
+                    }
+                    else
+                    {
+                        if (sInputObject_5009E8.isHeld(sInputKey_Up_4C6598))
+                        {
+                            field_FC_current_motion = eAbeStates::State_36_DunnoBegin_423260;
+                        }
+                    }
+                    return;
+                }
+                case TlvTypes::LocalWell_11:
+                {
+                    auto well = static_cast<Path_Well_Local*>(pTlv);
+                    if ((well->field_18_scale || field_BC_sprite_scale != FP_FromInteger(1)) &&
+                        (well->field_18_scale != 1 || field_BC_sprite_scale != FP_FromDouble(0.5)))
+                    {
+                        break;
+                    }
+                    field_2A8_flags.Clear(Flags_2A8::e2A8_Bit4_Fall_To_Well);
+                    field_F0_pTlv = pTlv;
+                    field_FC_current_motion = eAbeStates::State_77_WellBegin_430F10;
+                    return;
+                }
+                case TlvTypes::WellExpress_45:
+                {
+                    auto well = static_cast<Path_Well_Express*>(pTlv);
+                    if ((well->field_18_scale || field_BC_sprite_scale != FP_FromInteger(1)) &&
```

Comment:
is AE like this? I think its inverted in AE, checking for == 0 and == 1 is much nicer also

Reply:
Yes, AE is like this.